File: App_Code/RegExp.vb
Imports Microsoft.VisualBasic

Public Class RegExp

    Public Shared Function AlphaNumericValidation(ByVal IsMandatory As Boolean, ByVal MaxLength As Integer) As String
        Return "^[a-zA-Z0-9]{" & IIf(IsMandatory, "1", "0") & "," & MaxLength.ToString & "}$"
    End Function

    Public Shared Function EmailValidation(ByVal IsMandatory As Boolean, ByVal MaxLength As Integer) As String
        Return "^[a-zA-Z0-9 /.,\\\-_@]{" & IIf(IsMandatory, "1", "0") & "," & MaxLength.ToString & "}$"
    End Function

    Public Shared Function TextValidation(ByVal IsMandatory As Boolean, ByVal MaxLength As Integer) As String
        Return "^[a-zA-Z0-9 &%$#!*()+=\[\]:;<>?/.,\\\-_@']{" & IIf(IsMandatory, "1", "0") & "," & MaxLength.ToString & "}$"
    End Function

End Class

File: App_Code/RegExp.cs
public class RegExp
{
    public static string AlphaNumericValidation(bool IsMandatory, int MaxLength)
    {
        return "^[a-zA-Z0-9]{" + (IsMandatory ? "1" : "0") + "," + MaxLength.ToString() + "}$";
    }

    public static string EmailValidation(bool IsMandatory, int MaxLength)
    {
        return "^[a-zA-Z0-9 /.,\\\\\\-_@]{" + (IsMandatory ? "1" : "0") + "," + MaxLength.ToString() + "}$";
    }

    public static string TextValidation(bool IsMandatory, int MaxLength)
    {
        return "^[a-zA-Z0-9 &%$#!*()+=\\[\\]:;<>?/.,\\\\\\-_@']{" + (IsMandatory ? "1" : "0") + "," + MaxLength.ToString() + "}$";
    }
}
